### PR TITLE
Create a separate Group provider for Solaris systems.

### PR DESCRIPTION
**Pull request by jjustice6**
### Description
This creates a new Group provider for Solaris systems (`solaris2` environments), as the underlying `usermod` command behaves differently than how it does on other systems due to the +/- flag preceding the group list. See [the manpage](https://docs.oracle.com/cd/E26502_01/html/E29031/usermod-1m.html) for details.

### Issues Resolved

https://github.com/chef/chef/issues/8161

### Check List

- [X] New functionality includes tests
- [X] All tests pass
- [ ] RELEASE\_NOTES.md has been updated if required (not required for bugfixes, required for API changes)
- [X] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

**Comment by jjustice6** [comment]
I'm struggling to figure out exactly what needs to be where for the "other" solaris platforms.
Omnios points to http://wiki.omniosce.org/ManSections.html which suggests http://illumos.org/man/1M/usermod is the non +/- behavior.
SmartOS also has the other behavior. https://smartos.org/man/1M/usermod
OpenIndiana points to http://docs.openindiana.org/ which points to the Illumos pages.

So far I only see that actual Oracle Solaris 5.11 with the new behavior using the +/- on usermod.

So what's the best way to define the provides for this?

**Comment by lamont-granquist** [comment]
My $0.02 is to retain the existing behavior unless you can find something fairly definitive stating that the other solaris platforms use the new behavior.

Since none of them appear to use the new behavior, then the only provides like that you'd setup here would be for "solaris2" itself, and the others would all default to the old behavior.

Then we ensure we don't break anyone that isn't already broken.   For the people that are already broken it is up to them to do the research and submit the PR to change the behavior to fix it.

**Comment by jjustice6** [comment]
That makes sense to me.

File: lib/chef/provider/group/solaris.rb [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+#
+
+require "chef/provider/group/groupadd"
+
+class Chef
+  class Provider
+    class Group
+      class Solaris < Chef::Provider::Group::Groupadd
+
+        provides :group, platform: "solaris2"
```
Comment (tas50) [inline lib/chef/provider/group/solaris.rb]:
```suggestion
        # this provides line is setup to only catch the solaris2 platform and not other platforms in the Solaris platform_family      
        provides :group, platform: "solaris2"
```

File: spec/unit/provider_resolver_spec.rb [hunk 2]
```diff
@@ -851,8 +851,7 @@ def stub_service_configs(*configs)
               user: [ Chef::Resource::User::SolarisUser, Chef::Provider::User::Solaris ],
               "5.11" => {
```
Comment (tas50) [inline spec/unit/provider_resolver_spec.rb]: Is there an omnios or smartos spec in the provider_resolver spec so we can make sure the existing group provider still loads on these platforms?

File: lib/chef/provider/group/solaris.rb [hunk 3]
```diff
@@ -0,0 +1,72 @@
+#
+# Author:: Joshua Justice (<jjustice6@bloomberg.net>)
+# Copyright:: Copyright 2008-2016, Chef Software Inc.
```
Comment (tas50) [inline lib/chef/provider/group/solaris.rb]:
```suggestion
# Copyright:: Copyright 2008-2019, Chef Software Inc.
```

File: spec/unit/provider/group/solaris_spec.rb [hunk 4]
```diff
@@ -0,0 +1,105 @@
+#
+# Author:: Joshua Justice (<jjustice6@bloomberg.net>)
+# Copyright:: Copyright 2008-2016, Chef Software Inc.
```
Comment (tas50) [inline spec/unit/provider/group/solaris_spec.rb]:
```suggestion
# Copyright:: Copyright 2008-2019, Chef Software Inc.
```

File: spec/unit/provider_resolver_spec.rb (hunk 2 above)
Comment (jjustice6) [reply spec/unit/provider_resolver_spec.rb]: Yeah, there is.

File: lib/chef/provider/group/solaris.rb (hunk 1 above)
Comment (jjustice6) [reply lib/chef/provider/group/solaris.rb]: Tweaked comment further and did a separate commit so i could sign-off-by without having to do a force push like the others.

**Comment by tas50** [comment]
@jjustice6 Hey we merged out of order so you'll need to rebase this one now. I nuked some opensuse junk in #8158

File: spec/unit/provider_resolver_spec.rb (hunk 2 above)
Comment (tas50) [reply spec/unit/provider_resolver_spec.rb]: 👍

**Comment by jjustice6** [comment]
Is there anything else I need to do or any concerns I need to address?

**Comment by tas50** [comment]
@scotthain you have some Solaris experience can you take a peak at this?

**Pull request merged**